shared/common-adapters/popup-menu.desktop.js

```diff
@@ -53,8 +53,8 @@ class ModalLessPopupMenu extends Component<ModalLessPopupMenuProps> {
                   className={hoverClassName}
                   style={{...stylesRow, ...styleClickable}}
                   onClick={event => {
-                    i.onClick && i.onClick()
                     this.props.closeOnClick && this.props.onHidden && this.props.onHidden()
+                    i.onClick && i.onClick()
```

Comment:
why is this necessary

Reply:
mmmm, wonder what'll break if we make this change. If you use the navigateUp that comes from routeTree it should do nothing if the route has changed

Reply:
the issue wasn't a race, but that the callback was doing a `navigateAppend` on the dropdown route when it was only a child of the dropdown's parent. I could change it so the callback navigates up before appending to avoid making this change, but when we figure out a better popup solution I think the contract should be that the dropdown closes before callbacks are called (cc @akalin-keybase, if you're still interested in that ticket).